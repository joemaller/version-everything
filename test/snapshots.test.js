const path = require("path");
const fs = require("fs-extra");
const glob = require("glob");
const raw = require("jest-snapshot-serializer-raw").wrap;
const updateFile = require("../app/update-file");

/**
 * This file builds a snapshot test for every file in the **test-files**
 * directory. Snapshots are stored in `test/__snapshots__`.
 */

const testFiles = glob.sync("**/*", {
  cwd: path.resolve(__dirname, "snapshots"),
  nodir: true,
  absolute: true,
});

describe("Snapshot Tests:", () => {
  const newVersion = "1.414.213";

  testFiles.forEach((file) => {
    test(path.basename(file), () => {
      const src = fs.readFileSync(file).toString();

      return updateFile(file, newVersion, { dryRun: true, quiet: true }).then(
        (result) => {
          result = result || {};
          result.data = result.data || src;
          expect(raw(createSnapshot(src, result.data))).toMatchSnapshot();
          // done();
        }
      );

      // updateFile(file, newVersion, { dryRun: true }, (err, result) => {
      //   result = result || {};
      //   result.data = result.data || src;
      //   expect(raw(createSnapshot(src, result.data))).toMatchSnapshot();
      //   done();
      // });
    });
  });
});

function separator(width, title) {
  title = title || "";
  const leftLen = Math.floor((width - title.length) / 2);
  const rightLen = width - leftLen - title.length;
  return `${"=".repeat(leftLen)} ${title} ${"=".repeat(rightLen)}`;
}

function createSnapshot(input, output, options) {
  const width = 80;
  return [
    separator(width, "input"),
    input,
    separator(width, "output"),
    output,
  ].join("\n");
}
