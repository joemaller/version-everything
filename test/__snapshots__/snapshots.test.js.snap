// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests: Dockerfile 1`] = `
===================================== input ======================================
FROM alpine

LABEL version="1.23.456"

===================================== output =====================================
FROM alpine

LABEL version="1.414.213"

`;

exports[`Snapshot Tests: Dockerfile-indented-LABEL 1`] = `
===================================== input ======================================
  FROM alpine:latest

       LABEL version="0.444.1"

===================================== output =====================================
  FROM alpine:latest

       LABEL version="1.414.213"

`;

exports[`Snapshot Tests: Dockerfile-quoted-key 1`] = `
===================================== input ======================================
FROM alpine:latest

LABEL "version"="0.5555.1"

===================================== output =====================================
FROM alpine:latest

LABEL "version"="1.414.213"

`;

exports[`Snapshot Tests: cdata.xml 1`] = `
===================================== input ======================================
<?xml version="1.0"?>
<collection>
    <![CDATA[
   Version: 6.7.8
]]>
    <item >
        <taxonomy type="tag">
            <tag>food</tag>
            <tag>wine</tag>
        </taxonomy>
    </item>
</collection>

===================================== output =====================================
<?xml version="1.0"?>
<collection>
  <![CDATA[
   Version: 1.414.213
]]>
  <item>
    <taxonomy type="tag">
      <tag>food</tag>
      <tag>wine</tag>
    </taxonomy>
  </item>
</collection>
`;

exports[`Snapshot Tests: cdata-oneline.xml 1`] = `
===================================== input ======================================
<?xml version="1.0"?>
<collection>
    <![CDATA[ Version: 6.7.8 ]]>
    <item >
        <taxonomy type="tag">
            <tag>food</tag>
            <tag>wine</tag>
        </taxonomy>
    </item>
</collection>

===================================== output =====================================
<?xml version="1.0"?>
<collection>
  <![CDATA[ Version: 1.414.213 ]]>
  <item>
    <taxonomy type="tag">
      <tag>food</tag>
      <tag>wine</tag>
    </taxonomy>
  </item>
</collection>
`;

exports[`Snapshot Tests: decoy-version.md 1`] = `
===================================== input ======================================
# Version decoy file, these are not valid SemVer versions

## Version: 0

### Version 0.0

===================================== output =====================================
# Version decoy file, these are not valid SemVer versions

## Version: 0

### Version 0.0

`;

exports[`Snapshot Tests: sample.xml 1`] = `
===================================== input ======================================
<?xml version="1.0"?>
<catalog>
    <version>10.20.30</version>
    <!-- From https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms762271(v=vs.85) -->
    <book id="bk101">
        <author>Joe</author>
        <title>XML Test File</title>
    </book>
</catalog>

===================================== output =====================================
<?xml version="1.0"?>
<catalog>
  <version>1.414.213</version>
  <!-- From https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms762271(v=vs.85) -->
  <book id="bk101">
    <author>Joe</author>
    <title>XML Test File</title>
  </book>
</catalog>
`;

exports[`Snapshot Tests: simple-package.json 1`] = `
===================================== input ======================================
{
  "version": "1.22.333"
}

===================================== output =====================================
{
  "version": "1.414.213"
}
`;

exports[`Snapshot Tests: w3schools-note.xml 1`] = `
===================================== input ======================================
<?xml version="1.0" encoding="UTF-8"?>
<note>
  <to>Tove</to>
  <from>Jani</from>
  <heading>Reminder</heading>
  <body>Don't forget me this weekend!</body>
</note>

===================================== output =====================================
<?xml version="1.0" encoding="UTF-8"?>
<note>
  <version>1.414.213</version>
  <to>Tove</to>
  <from>Jani</from>
  <heading>Reminder</heading>
  <body>Don't forget me this weekend!</body>
</note>
`;

exports[`Snapshot Tests: w3schools-simple.xml 1`] = `
===================================== input ======================================
<?xml version="1.0" encoding="UTF-8"?>
<breakfast_menu>
  <food>
    <name>Belgian Waffles</name>
    <price>$5.95</price>
    <description>Two of our famous Belgian Waffles with plenty of real maple syrup</description>
    <calories>650</calories>
  </food>
  <food>
    <name>Strawberry Belgian Waffles</name>
    <price>$7.95</price>
    <description>Light Belgian waffles covered with strawberries and whipped cream</description>
    <calories>900</calories>
  </food>
  <food>
    <name>Berry-Berry Belgian Waffles</name>
    <price>$8.95</price>
    <description>Light Belgian waffles covered with an assortment of fresh berries and whipped cream</description>
    <calories>900</calories>
  </food>
  <food>
    <name>French Toast</name>
    <price>$4.50</price>
    <description>Thick slices made from our homemade sourdough bread</description>
    <calories>600</calories>
  </food>
  <food>
    <name>Homestyle Breakfast</name>
    <price>$6.95</price>
    <description>Two eggs, bacon or sausage, toast, and our ever-popular hash browns</description>
    <calories>950</calories>
  </food>
</breakfast_menu>

===================================== output =====================================
<?xml version="1.0" encoding="UTF-8"?>
<breakfast_menu>
  <version>1.414.213</version>
  <food>
    <name>Belgian Waffles</name>
    <price>$5.95</price>
    <description>Two of our famous Belgian Waffles with plenty of real maple syrup</description>
    <calories>650</calories>
  </food>
  <food>
    <name>Strawberry Belgian Waffles</name>
    <price>$7.95</price>
    <description>Light Belgian waffles covered with strawberries and whipped cream</description>
    <calories>900</calories>
  </food>
  <food>
    <name>Berry-Berry Belgian Waffles</name>
    <price>$8.95</price>
    <description>Light Belgian waffles covered with an assortment of fresh berries and whipped cream</description>
    <calories>900</calories>
  </food>
  <food>
    <name>French Toast</name>
    <price>$4.50</price>
    <description>Thick slices made from our homemade sourdough bread</description>
    <calories>600</calories>
  </food>
  <food>
    <name>Homestyle Breakfast</name>
    <price>$6.95</price>
    <description>Two eggs, bacon or sausage, toast, and our ever-popular hash browns</description>
    <calories>950</calories>
  </food>
</breakfast_menu>
`;
